src/uwtools/drivers/make_hgrid.py

```diff
@@ -0,0 +1,72 @@
+"""
+A driver for make_hgrid.
+"""
+
+from pathlib import Path
+from typing import Optional
+
+from iotaa import asset, dryrun, task, tasks
+
+from uwtools.drivers.driver import Driver
+from uwtools.strings import STR
+
+
+class MakeHgrid(Driver):
+    """
+    A driver for make_hgrid.
+    """
+
+    def __init__(
+        self,
+        config: Optional[Path] = None,
+        dry_run: bool = False,
+        batch: bool = False,
+    ):
+        """
+        The driver.
+
+        :param config: Path to config file (read stdin if missing or None).
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        super().__init__(config=config, dry_run=dry_run, batch=batch)
+        if self._dry_run:
+            dryrun()
+
+    # Workflow tasks
+
+    @tasks
+    def provisioned_run_directory(self):
+        """
+        Run directory provisioned with all required content.
+        """
+        yield self._taskname("provisioned run directory")
+        yield self.runscript()
+
+    @task
+    def runscript(self):
+        """
+        The runscript.
+        """
+        path = self._runscript_path
+        yield self._taskname(path.name)
+        yield asset(path, path.is_file)
+        yield None
+        self._write_runscript(path=path, envvars={})
```

Comment:
@christinaholtNOAA From your question, this ~should~ be converting the YAML keys to vars to write, but it doesn't appear to be actually doing that, let me retest this

Reply:
Update: It was in the` run_cmd` override we forgot and removed earlier, thank you! Also updated to correctly handle the keys and their conditions